lucene/core/src/java/org/apache/lucene/search/SortField.java

```diff
@@ -408,6 +428,7 @@ public boolean equals(Object o) {
       && other.reverse == this.reverse
       && Objects.equals(this.comparatorSource, other.comparatorSource)
       && Objects.equals(this.missingValue, other.missingValue)
+      && other.skipNonCompetitiveDocs == this.skipNonCompetitiveDocs
```

Comment:
I wonder if we should actually take it into account for equals/hashcode as this could disable index sorting optimizations, which check for equality of the index-time and search-time SortField instances.

Reply:
@jpountz Thank you for your feedback. Alternatively we can add `setCanUsePoints` parameter to `Sort` object instead of `SortField`?  What do you think?